### ISSUE TITLE
Tracy Profiler crash on startup in Worker mutex lock with access violation on Windows

### ISSUE THREAD

**Issue by egorodet** [wolfpld/tracy#887]
`tracy-profiler.exe` downloaded from GitHub releases v0.11.0 and v0.11.1 does not start silently.

When I build Tracy Profiler from sources, it works normally in `Debug` build, but crashes in `Release` and `ReleaseWithDebInfo` builds on attempt to lock mutex from thread loop of `ResolvService::Worker()`. Issue is reproducible on `master`, `v0.11.1` and `v0.11.0`. Previous release of Tracy `v0.10` was working fine.

![Tracy-Mutex-Crash-2](https://github.com/user-attachments/assets/db9fe796-998f-4ff8-b364-21cafdfffa1e)

![Tracy-Mutex-Crash-1](https://github.com/user-attachments/assets/9ed5a5c6-9611-44d6-b9eb-6c0f99aff0c1)

**Build command line**:
```cmd
cd X:\Git\Tracy\profiler
cmake -S . -B Build -G "Visual Studio 17 2022" -A x64
cmake --build Build --config Release
```

**System Information**:
- Windows 11 Pro 23H2 22631.4169
- Microsoft Visual Studio Enterprise 2022 v17.11.3

**Comment by wolfpld** [comment]
See #862. The fact that you have MSVC 17.11 doesn't fit though.

**Comment by egorodet** [comment]
> See #862. The fact that you have MSVC 17.11 doesn't fit though.

Oh, now I see. Looks like a new disaster :((( I though that MSVC Runtime is auto-updated by Visual Studio Installer along with other updates, but seems like it does not... 

Maybe instead of updating runtimes, it would be better to add definition `_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR` (as [suggested by Microsoft](https://developercommunity.visualstudio.com/t/Visual-Studio-17100-Update-leads-to-Pr/10669759?sort=newest)) to return compatibility with old runtime versions?

**Comment by egorodet** [comment]
Btw, I've downloaded latest MSVC runtime installer `v14.40.33810.0` [from here](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022) and it says that it is already installed! What's next?

![image](https://github.com/user-attachments/assets/b2e1e105-9c99-4a9f-bd6f-73ef6f2513ac)

**Comment by wolfpld** [comment]
> Oh, now I see. Looks like a new disaster :((( I though that MSVC Runtime is auto-updated by Visual Studio Installer along with other updates, but seems like it does not...

Oh wow, so you can end up with a MSVC that produces binaries that can run? That's a new low, even for MSVC.

> Maybe instead of updating runtimes, it would be better to add definition `_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR` (as [suggested by Microsoft](https://developercommunity.visualstudio.com/t/Visual-Studio-17100-Update-leads-to-Pr/10669759?sort=newest)) to return compatibility with old runtime versions?

Yeah, this felt super-hacky to me, but considering the new circumstances, that may be the right way to go.

7792b954 have this enabled.

**Comment by egorodet** [comment]
> [7792b95](https://github.com/wolfpld/tracy/commit/7792b95443633bb9b8a18c09834fdaefbdcbb1c2) has this enabled.

Yep, with this definition added, Tracy Profiler runs perfectly fine with MSVC runtime unchanged!

![image](https://github.com/user-attachments/assets/ffee43f4-5be8-44ff-9069-88b0bc29c927)

**Comment by mattico** [comment]
I ran into this issue as well. I definitely have the most recent MSVCRT installed, and even my locally built versions of tracy would *sometimes* have this issue.

I think I figured out the cause using Dependency Walker. The exe is loading `api-ms-win-core-*.dll` from Microsoft's JDK, which inserted itself into the PATH before C:\Windows, etc. The JDK evidently bundles an old version of MSVCRT. I suspect something similar is the cause of the issue for anyone who has the latest MSVCRT installed.

I'm not sure what the "proper" fix is supposed to be for this issue. They seem to discourage bundling those `api-ms-win-core-*.dll`, and it looks like the app is only supposed to go searching for DLLs in PATH as a last resort if it doesn't find the DLL in PWD or C:\Windows or WinSxS, or whatever. Anyway the workaround works fine.
